HeterogeneousCore/SonicTriton/interface/TritonService.h

```diff
@@ -132,6 +134,7 @@ class TritonService {
   unsigned currentModuleId_;
   bool allowAddModel_;
   bool startedFallback_;
+  CMS_SA_ALLOW mutable std::atomic<int> callFails_;
```

Comment:
Wouldn't `bool` along
```suggestion
  CMS_SA_ALLOW mutable std::atomic<bool> anyCallFailed_{false};
```
be sufficient? (only set to `true` if any of the inference calls failed)

Reply:
Whenever a call fails because of a potential local fallback server issue, a `TritonException` is created with the instruction to notify the `TritonService` that the call failed. But calls can be retried, and when a call is retried, it has to undo that notification. Only allowing `true` and `false` as possibilities means that the notification from a completely-failed call could get overwritten by a retrying call (depending on how these notifications are interleaved when running on multiple threads). Maybe there is a smarter way to do this, but an atomic counter seemed appropriate here.

Reply:
Ok. I forgot earlier to mention the `CMS_SA_ALLOW` is not needed for `std::atomic` (because static analyzer knows `std::atomic` is fully thread safe)
```suggestion
  mutable std::atomic<int> callFails_;
```